runtime/doc/treesitter.txt

```diff
@@ -292,7 +292,39 @@ The following directives are built in:
         Example: >query
             ((identifier) @constant (#offset! @constant 0 1 0 -1))
 <
+    `gsub!`                                          *treesitter-directive-gsub!*
```

Comment:
It would be good if we could steer our directives so they are as digestible as possible by other TS implementations (for the very long term), so I'd prefer if there was a way to not include this.

The others added here should be safe.

Reply:
I just added the documentation for the already existing (and used) directive. Can you explain what you mean by "digestible" and "TS implementations"?

Reply:
> Not sure when or why it was added.

https://github.com/neovim/neovim/pull/21548 (recall that you are interested in that precise use case)

Reply:
> And with that argument, we'd also have to remove our #lua-match? and #match? ( == #vim-match?) predicates.

Yes, eventually and replace with something that covers a subset. I thought we talked about this?

> I'm sorry, but that makes no sense; all of our directives necessarily require Lua, as this is what they are 

`trim`, `set`, `offset`do not require Lua. They can be trivially implemented in any language.

Reply:
> Yes, eventually, and replace with something that covers a subset. I thought we talked about this?

Not that I recall (or at least I don't recall such a strong conclusion). In any case, since the mere (even undocumented) presence of the directive has side effects (an error if a user/plugin tries to register the same directive), I feel strongly that the choice should be all-or-nothing here. Given that there's already precedent with `#lua-match!` and directives are explicitly documented as "downstream responsibility" by tree-sitter, my vote would be to keep it in (and possibly replace it in the same far future PR that replaces `#lua-match!`, given that both rely on Lua patterns).

Reply:
But I have no skin in the game otherwise; yeeting it is OK with me, too. (But then it'll stay gone; I'm not keen in shuffling this back to nvim-treesitter.)